src/rdkafka_broker.c

```diff
@@ -3614,20 +3635,58 @@ rd_kafka_fetch_reply_handle (rd_kafka_broker_t *rkb,
                                 int32_t MessageSetSize;
                         } hdr;
                         rd_kafka_resp_err_t err;
+                        int64_t end_offset;
 
 			rd_kafka_buf_read_i32(rkbuf, &hdr.Partition);
 			rd_kafka_buf_read_i16(rkbuf, &hdr.ErrorCode);
 			rd_kafka_buf_read_i64(rkbuf, &hdr.HighwaterMarkOffset);
 
+                        end_offset = hdr.HighwaterMarkOffset;
+
                         if (rd_kafka_buf_ApiVersion(request) == 4) {
                                 int32_t AbortedTxCnt;
                                 rd_kafka_buf_read_i64(rkbuf,
                                                       &hdr.LastStableOffset);
-                                rd_kafka_buf_read_i32(rkbuf, &AbortedTxCnt);
-                                /* Ignore aborted transactions for now */
-                                if (AbortedTxCnt > 0)
-                                        rd_kafka_buf_skip(rkbuf,
-                                                          AbortedTxCnt * (8+8));
+                                rd_kafka_buf_read_i32(rkbuf,
+                                                      &AbortedTxCnt);
+
+                                if (rkb->rkb_rk->rk_conf.isolation_level ==
+                                    RD_KAFKA_READ_COMMITTED)
+                                        end_offset = hdr.LastStableOffset;
+
+                                if (AbortedTxCnt > 0) {
+                                        if (unlikely(rkb->rkb_rk->rk_conf.isolation_level !=
+                                            RD_KAFKA_READ_COMMITTED))
+                                                rd_kafka_buf_parse_fail(
+                                                        rkbuf,
+                                                        "%.*s [%"PRId32"]: "
+                                                        "Unexpected aborted "
+                                                        "transaction count %"PRId32,
+                                                        RD_KAFKAP_STR_PR(&topic),
+                                                        hdr.Partition,
+                                                        AbortedTxCnt);
+
+                                        abrt = rd_malloc(sizeof(rd_kafka_abrt_trans_state_t));
+                                        
+                                        abrt->list_idx = 0;
+                                        
+                                        rd_list_init(&abrt->trans_list, 0, NULL);
```

Comment:
i think the word list here adds value since it adds clarity to anyone following along with the design doc. however, as you point out offline, we're doing to implement this differently anyway. changed trans -> txn.